### PR TITLE
Added S3 DELETE request handler

### PR DESCRIPTION
**Pull request by weisong44**
As per subject, added S3 DELETE request handler, initial implementation includes

1. delete object handler according to [AWS spec](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html)

### CODE REVIEW

File: ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3DeleteHandler.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    String path = ((RequestPath) restRequest.getArgs().get(REQUEST_PATH)).getOperationOrBlobId(true);
+    LOGGER.debug("Delete {}", path);
+
+    if (!path.startsWith(Operations.NAMED_BLOB)) {
+      throw new RuntimeException("S3DeleteHandler only handles named blob requests");
+    }
+
+    Callback<ReadableStreamChannel> wrappedCallback = (result, exception) -> {
+
+      // TODO: remove x-ambry- headers
```
Comment (JingQianCloud) [inline ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3DeleteHandler.java:68]: TODO [S3]

**Review by JingQianCloud — APPROVED** [review]
LGTM

**Review by Arun-LinkedIn — APPROVED** [review]
Looks good to me!

File: ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3DeleteHandler.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   * Process the request and construct the response according to the
+   * <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html">DeleteObject spec</a> and
+   *
+   * @param restRequest the {@link RestRequest} that contains the request parameters and body.
+   * @param restResponseChannel the {@link RestResponseChannel} where headers should be set.
+   * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+   * @throws RestServiceException exception when the processing fails
+   */
+  public void handle(RestRequest restRequest, RestResponseChannel restResponseChannel,
+      Callback<ReadableStreamChannel> callback) throws RestServiceException {
```
[the remark this answers is not in the document]
Comment (weisong44) [reply ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3DeleteHandler.java:58]: Good catch, fixed!

File: ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3DeleteHandler.java (hunk 1 above)
Comment (weisong44) [reply ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3DeleteHandler.java:68]: Fixed

**Pull request merged**